src/migrator/utils.ts

```diff
@@ -105,6 +105,12 @@ export type MigrationStatus = {
 	is_backing_off: boolean;
 };
 
+export type MigrationExecutionResult =
+	| undefined
+	| {
+			pendingUnsetMigrations?: string[];
```

Comment:
Overall I just wonder if it woudln't be better to make the whole typing in here more strict e.g.
```
export type MigrationExecutionResult = {
    pendingUnsetMigrations: string[];
}
```

And then adapting the functions to return something explicetely saying an empty list of pendingUnsetMigrations rather than just undefined

Reply:
I'm going to use a strict `pendingUnsetMigrations: string[];` but will continue use the `undefined` return when there is nothing. 
In the current implementation it would cause 'unnecessary' empty object returns